src/modules/core/components/NetworkStatus/NetworkStatus.tsx

```diff
@@ -90,7 +96,16 @@ const NetworkStatus = () => {
       }
     }, networkCheckInterval);
     return () => clearInterval(networkCheck);
-  });
+  }, [
+    latestRpcBlockError,
+    isReputationOracleAlive,
+    isColonyServerAlive,
+    ipfsLivenessData,
+    latestRpcBlock,
+    latestSubgraphBlock,
+    latestSubgraphBlockError,
+    networkCheckInterval,
+  ]);
```

Comment:
Are you sure about this dependency array? IIUC, it means if one of those requests take some time and changes one of these things, the setInterval starts afresh. This means in practice there are way more frequent requests to the server than the 5 minutes interval (which puts extra pressure on the server).

Reply:
Yes, I'm sure. This `useEffect` is not making any requests to the networks, it's only modifying the local state of `networkHealth` via `setNetworkHealth` so no requests are made. 

Also now that you mention this, I took a closer look at the interval and I believe that it can be safely removed now that we have a dependency array, it should also be better in the sense that before without the dependency array and the interval the useEffect would not know of any updated values

Reply:
Ummm, I am still not convinced about this. So, are you guaranteeing that the checks will demonstrably fire at every 5 minutes **only**? Last time I checked they were firing much more frequently than that (due to the reason I mentioned above. Maybe I am missing something. If you are confident then let me know and I will approve this.

Reply:
If you are so unsure about it you don't need to rely on my words to know if this works like I'm saying it does or not, we can _actually_ test it:

Placed some `console.log()` inside this `useEffect` and each of the checks we do:
![Screenshot from 2022-05-04 12-21-31](https://user-images.githubusercontent.com/18473896/166716268-12ac1fc2-b0e3-4a59-aead-3b760ff4c41b.png)

This is the initial load, component renders, `useEffect` and queries are executed for the first time. After some of the queries return the data the `useEffect` is executed once more because the values were updated by the queries.

Again a few seconds later the useEffect ran again because some of the above queries probable finished returning data:
![Screenshot from 2022-05-04 12-18-19](https://user-images.githubusercontent.com/18473896/166717536-81ebf378-4b3a-4c76-934c-0d9724177022.png)

Finally, nothing more happens until _5 minutes_ later when the queries ran again due to the polling interval that they have: 
![Screenshot from 2022-05-04 12-23-41](https://user-images.githubusercontent.com/18473896/166717994-f9a7294d-24d7-4963-adcb-3b754ab60efb.png)

BUT the `useEffect` didn't run again because the values of the queries didn't change since the last time it was queried.

So, `useEffect` running 6 times in total as of now doesn't affect the number of queries because the `useEffect` itself is not executing any queries, the `useEffect` is only doing state management (setting network health).

Hope this clears any doubts.

Reply:
I missed the polling logic. And with the removal of the interval I think it works as expected. Great work!